helm/cloud-provider-azure/templates/cloud-provider-azure.yaml

```diff
@@ -428,6 +428,21 @@ spec:
               cpu: {{ .Values.cloudNodeManager.containerResourceManagement.limitsCPU }}
               memory: {{ .Values.cloudNodeManager.containerResourceManagement.limitsMem }}
 ---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: cloud-provider-azure-scripts
+  namespace: kube-system
+  labels:
+    app: cloud-provider-azure-windows
+data:
+  start.ps1: |
+    $cm = "$env:CONTAINER_SANDBOX_MOUNT_POINT/cloud-node-manager.exe"
+
+    ((Get-Content -path $env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf -Raw) -replace '/var',"$($env:CONTAINER_SANDBOX_MOUNT_POINT)/var") | Set-Content -Path $env:CONTAINER_SANDBOX_MOUNT_POINT/var/lib/cpaw/kubeconfig.conf
```

Comment:
Could the in-cluster authz with service account work here? If so, there is no need of the script

Reply:
In-cluster auth with service accounts works for hostprocess containers with containerd v1.7 only (not v1.6.x).
The way volume mounts get set up was changed between containerd 1.6 and 1.7

Reply:
yes this is a workaround until containerd 1.7 is released, after that we can remove the script

Reply:
then let's wait for 1.7?

Reply:
it's blocking https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3105 

we want to move all CAPZ templates and tests to external cloud-provider to get as much test coverage as possible before in-tree cloud-providers are completely removed

Reply:
I see. Could you add a TODO comment with containerd issue link?

Reply:
Here is the issue https://github.com/kubernetes/kubernetes/issues/104562

Reply:
Added